### PR TITLE
Rule chain implementation for contagion checks

### PR DESCRIPTION
**Pull request by NachoRaik**
### Description

This PR adds the logic of a simple (non-concurrent) rule chain, for processing possible compromised user codes. This includes:
- Whole project structure for this Go repo
- DB and RabbitMQ integration
- Rule chain based on code batches for spaces, where each step of the chain is a particular Rule check
- A result handler that writes the resulting compromised codes to the DB, if a match is found in any of the chain's Rule checks
- The batch processing is scheduled to be run every certain amount of time (for the sake of testing and development, it is now set to every 20 seconds)

----------------------------------------------------

##### Further TODOs

Some of them are written in the code itself, but these include:
- Improve a little bit the project structure for some parts (ex: models)
- Standardize logging (currently, there are alternates uses of `fmt` and `log`) --> Investigate logging libraries
- Try using pointers in the `Rule` struct, to avoid doing existence checks like `!=0` or `!=""`
- Separate current structs into DTOs and internal model structs (that use agnostic types, for example)
- Parameterize scheduled job period via env var

##### Further Improvements

- Concurrent Rule Chain
- Batch Result Handler

### CODE REVIEW

**Review by tomasaccini — COMMENTED** [review]
Increible laburo :) un par de comentarios abajo. Hay partes que no me terminan de quedar del todo claras, capaz me vendria bien una meeting cortita para que me expliques un toque, pero si oli lo entiende y da el ok mergeemos

**Review by tomasaccini — APPROVED** [review]
Aprobando despues de la reunion explicando todo! Agrego al backlog lo que habiamos hablado de setear una variable de entorno para que se pueda correr el worker con cada codigo que llega asi es mas facil armar test automaticos sin tener que poner sleeps en el medio

File: cmd/main.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+	dbUri := os.Getenv("MONGODB_URI")
+	dbName := os.Getenv("MONGODB_DB_NAME")
+
+	db, err := mongodb.New(dbUri, dbName)
+	logger.FailOnError(err, "Failed to connect to the DB")
+	defer db.Shutdown()
+
+	consumer, err := rabbitmq.New(queueAddress, queueName)
+	logger.FailOnError(err, "Failed to register a consumer")
+	defer consumer.Shutdown()
```
[the remark this answers is not in the document]
Comment (NachoRaik) [reply cmd/main.go:31]: Se hace porque el `defer` es justamente ejecutar la acción al salir del scope, pase lo que pase

File: pkg/controllers/infected_manager.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
+package controllers
+
+import (
+	"fmt"
+
+	"github.com/ct-fiuba/ct-contagion-updater/pkg/models/compromisedCodes"
+	"github.com/ct-fiuba/ct-contagion-updater/pkg/models/rules"
+	"github.com/ct-fiuba/ct-contagion-updater/pkg/models/spaces"
+	"github.com/ct-fiuba/ct-contagion-updater/pkg/models/visits"
+	// rd_api "github.com/ct-fiuba/ct-contagion-updater/pkg/riskdetecter/api"
```
Comment (olifer97) [inline pkg/controllers/infected_manager.go:10]: remove?

File: pkg/riskdetecter/impl/simple_rule_chain.go [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+	// 		EntranceTimestamp: primitive.NewDateTimeFromTime(someTime.Add(-someDuration2)),
+	// 		Vaccinated:        0,
+	// 		CovidRecovered:    false,
+	// 	},
+	// }
+	// for _, v := range relatedVisits {
+	// 	initialFilter := rulechain.filters[0].checker
+	// 	fmt.Printf("AAAAAAAAAA, %+v \n", initialFilter)
+	// 	initialFilter.Execute(infectedVisit, v)
+	// }
```
Comment (olifer97) [inline pkg/riskdetecter/impl/simple_rule_chain.go:80]: remove?

**Review by olifer97 — APPROVED** [review]
laburazo nach!

**Pull request merged**